### DISCUSSION TITLE
Some misunderstand about data and LaTeX codes

### DISCUSSION THREAD

**Discussion by tuiiitendinh** [lukas-blecher/LaTeX-OCR#189] [General]
Hi Lukas, I have some confusing about the dataset.
In the text file contains the LaTeX code, eg `CROHME_math.txt` or `pdfmath.txt`. 
There are spaces between each part of the LaTeX code. 
Compared to some other datasets, despite knowing that the spaces in the LaTeX code doesn't affect the whole equation when rendering it. But what's the purpose of having those spaces in the LaTeX code?
Thanks

**Comment by lukas-blecher** [comment]
The idea originates from the original harvard paper ([github](https://github.com/harvardnlp/im2markup), [arxiv](https://arxiv.org/abs/1609.04938v2)). The idea is that each command/character is done one at a time which helps in giving the model an underlying structure.
I don't think the spaces are really needed these times, but in the same step the latex code is also normalized (eg `\frac12` to `\frac{1}{2}`) which helps to reduce the complexity of the task in eliminating ambiguity.

The authors report
> Surprisingly we find this additional step gives only a small accuracy gain
